mmcv/engine/test.py

```diff
@@ -32,8 +32,7 @@ def single_gpu_test(model, data_loader):
             result = model(return_loss=False, **data)
         results.extend(result)
 
-        # use the first key as main key to calculate the batch size
-        batch_size = len(next(iter(data.values())))
```

Comment:
Add comment to indicate that: we assume the results has similar length as batch size, and refer to this PR and issue.